src/pages/iou/request/step/IOURequestStepMerchant.tsx

```diff
@@ -123,7 +97,7 @@ function IOURequestStepMerchant({
             navigateBack();
             return;
         }
-        IOU.setMoneyRequestMerchant(transactionID, newMerchant, !isEditing);
+        IOU.setMoneyRequestMerchant(transactionID, newMerchant ?? '', !isEditing);
```

Comment:
This empty string fallback caused issue https://github.com/Expensify/App/issues/41939 where if we set the merchant to something else first, for example "test", then the transaction draft merchant becomes "test".
`merchant: 'test'`
Then if we clear it, becomes empty string.
`merchant: ''`

If we submit an expense with an empty string of merchant, the BE response will return `Expense` as the merchant.

**Note:** The fallback should've been `newMerchant || CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT` like it is below for `updateMoneyRequestMerchant` function.